app/controllers/users_controller.rb

```diff
@@ -18,6 +19,51 @@ def create
     end
   end
 
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+
+    # the edit view will return 'false'
+    if params[:changing_password] == 'false'
+      params[:user].delete :password
+      params[:user].delete :password_confirmation
+      if @user.update_attributes(user_params)
+        redirect_to @user
+      else
+        # rendering edit will change URL, possibly
+        # breaking button to 'Change password' in 'edit' view
+        # FIXME: correct URL in later version
```

Comment:
This does break the change password button if you try to update your profile with bogus info.